### PR TITLE
FSPT-598: load submission status for AGF grant page

### PR DESCRIPTION
**Pull request by samuelhwilliams**
Relies on #420 

## 🎫 Ticket
https://mhclgdigital.atlassian.net/browse/FSPT-598

## 📝 Description
Updates the Access grant funding 'Grant details' page to load the status of any submissions that exist for the user into the collections table.

### Before
image

### After
image

## 🧪 Testing
I created multiple collections, started one through Deliver grant funding, and then checked the grant details page on Access grant funding.

### Review Readiness
- [x] PR title and description provide sufficient context for reviewers
- [x] Commits are logical, self-contained, and have clear descriptions

### Testing
- [x] I have tested this change and it meets the acceptance criteria for the ticket
~- [ ] I need the reviewer(s) to pull and run this change locally~
~- [ ] New (non-developer) functionality has appropriate unit and integration tests~
~- [ ] End-to-end tests have been updated (if applicable)~
~- [ ] Edge cases and error conditions are tested~

### CODE REVIEW

File: app/developers/access_routes.py [hunk 1]
```diff
@@ -12,3 +18,22 @@
...
+#       anonymous vs logged-in user.
+@developers_access_blueprint.get("/grant/<uuid:grant_id>")
+def grant_details(grant_id: uuid.UUID) -> ResponseReturnValue:
+    grant = get_grant(grant_id)
+    submission_helpers = {
+        collection.id: SubmissionHelper.load(submission.id)
+        for collection in grant.collections
+        if (submission := get_submission_by_collection_and_user(collection, interfaces.user.get_current_user()))
+    }
+    # todo: feed submissions/submission_helpers into the template to render statuses
```
Comment (MarcUsher) [inline app/developers/access_routes.py:33]: Lil todo that can be removed now I think

**Review by MarcUsher — DISMISSED** [review]
Approving with just a minor catch of a todo that can be removed 🙏

**Review by MarcUsher — APPROVED** [review]
Nice, this works for me locally and the AnonymousUser behaviour is also behaving as expected 🙏

This feels like one we should give a heads up to for other devs ahead of us working on more publicly accessible pages (if we'll have many/any)
